### ISSUE TITLE
redirects in enwiki

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#443]
this is indirectly related to #442 
Looking at a page in enwiki that has a 'strange' redirect (in the database has page_is_redirect=1 and page_redirect=-1)
Specifically en.wikipedia.org/wiki/8D8
This gives
![8d8](https://user-images.githubusercontent.com/861631/56880335-2f9b1300-6a54-11e9-8ec1-76889faea182.jpg)
Clicking on the no redirect link
(which is en.wikipedia.org/wiki/8D8?redirect=no) gives
![8d8no](https://user-images.githubusercontent.com/861631/56880448-a2a48980-6a54-11e9-9101-b1c31e33bfb5.jpg)

In this specific case, the issue here are the two hash '#' characters
xowa has pealed one off and doesn't know of a page called 'List_of_Star_Wars_characters#'
mediawiki shows the page as
![8d8wiki](https://user-images.githubusercontent.com/861631/56880626-6b82a800-6a55-11e9-9061-e669bcfedaf6.jpg)
The hashes are an anchor link to a hash

This also highlights another small issue
There is no way to look at the wikitext of a redirect=no
(that is en.wikipedia.org/wiki/8D8?action=edit)
as it looks like the redirect occurs first

**Comment by desb42** [comment]
rather than a new issue - under the same heading this redirect
en.wikipedia.org/wiki/Piotr_nowina_konopka does follow through to a valid page
en.wikipedia.org/wiki/Piotr_Nowina-Konopka

The question is why no update of page_redirect_id?
Is it because the redirect

    REDIRECT en:Piotr_Nowina-Konopka
has `en:` on it?

**Comment by gnosygnu** [comment]
> In this specific case, the issue here are the two hash '#' characters
> xowa has pealed one off and doesn't know of a page called 'List_of_Star_Wars_characters#'

Yeah, this is a bug in the redirect parser. This was a piece I wrote without consulting MediaWiki code. You can see the main class here: https://github.com/gnosygnu/xowa/blob/341d2e13a43bb91e40b0c0204a750a5f56780a26/400_xowa/src/gplx/xowa/parsers/utils/Xop_redirect_mgr.java

The double hash should be easy to fix, but let me try to hunt down the MediaWiki code and see what else I'm missing

> en.wikipedia.org/wiki/Piotr_nowina_konopka does follow through to a valid page en.wikipedia.org/wiki/Piotr_Nowina-Konopka
> Is it because the redirect
> REDIRECT en:Piotr_Nowina-Konopka
> has en: on it?

Yup. Same thing. :(

> There is no way to look at the wikitext of a redirect=no
> (that is en.wikipedia.org/wiki/8D8?action=edit)
> as it looks like the redirect occurs first

Nice catch. Separate problem, but will fix under this issue.

**Comment by desb42** [comment]
I think I have found why the ## issue occurs
In 400_xowa\src\gplx\xowa\Xoa_ttl.java
Line 368 needs an additional check; that the anchor has not already been set

    if (b_ary.length == 1 && b_ary[0] == Byte_ascii.Hash)	// NOTE: A&#x23;B should be interpreted as A#b; PAGE:en.s:The_English_Constitution_(1894) DATE:2014-09-07
to

    if (b_ary.length == 1 && b_ary[0] == Byte_ascii.Hash && anch_bgn == -1)	// NOTE: A&#x23;B should be interpreted as A#b; PAGE:en.s:The_English_Constitution_(1894) DATE:2014-09-07
